### PR TITLE
[unique.ptr] Remove definition of 'transfers ownership'.

### LINKED ISSUE #1643: [unique.ptr]  Consider removing "transfer ownership" definition
All of `unique_ptr::operator=(unique_ptr&&)`, `shared_ptr(shared_ptr&&)`, and `shared_ptr::operator=(shared_ptr&&)` explicitly leave their source with `.get()==nullptr`. `unique_ptr(unique_ptr&&)`'s wording delegates that to the "transfer ownership" definition in [unique.ptr], without explicitly referring to the section of the definition, which caused me to miss that it was defined at all.

As far as I can see, that term is only used twice in the standard, for the `unique_ptr` move constructor and templated move constructor, and I believe its other 2 bullets are repeated by the existing postconditions for those operations. By adding `u.get()==nullptr` to those two sets of postconditions, we might be able to remove the whole definition of "transfer ownership".

### PR DESCRIPTION
**Pull request by jensmaurer**
It is mostly subsumed by the detailed descriptions of the move
constructors and assignment operators.

Fixes #1643.

### CODE REVIEW

**Comment by zygoloid** [comment]
I would like us to consider an alternative: retain the definition of transfer of ownership, and replace the repeated postconditions with

> \postcondition As specified for transfer of ownership from \tcode{foo} to \tcode{bar}\iref{unique.ptr}.

I'm weakly in favor of this alternative, to reduce duplication and give us somewhere central to talk about `unique_ptr`'s ownership semantics. However, if there's some reason the duplicated-postconditions approach is superior, I expect it would not take much to change my mind.

**Comment by jensmaurer** [comment]
So, we would substantially shorten the existing postconditions.

Hm.. The "transfer ownership" wording is imprecise regarding the deleter: It says "if the pre-transfer u.d maintained state, such state has been transferred to u2.d", but the move constructors actually say that we copy-constructor or move-construct the deleter. Oh, move-assignment actually shows code rather than weasely wording.

File: source/utilities.tex [hunk 1]
```diff
@@ -8316,15 +8299,16 @@
 throw an exception.
 
 \pnum
-\effects Constructs a \tcode{unique_ptr} by transferring ownership from
-\tcode{u} to \tcode{*this}. If \tcode{D} is a reference type, this
+\effects Constructs a \tcode{unique_ptr} from
+\tcode{u}. If \tcode{D} is a reference type, this
 deleter is copy constructed from \tcode{u}'s deleter; otherwise, this
 deleter is move constructed from \tcode{u}'s deleter. \begin{note} The
 deleter constructor can be implemented with \tcode{std::forward<D>}. \end{note}
```
Comment (zygoloid) [inline source/utilities.tex]: (Drive-by:) "The construction of the deleter" would be more correct here.

**Comment by zygoloid** [comment]
OK. The idea of using "transfers ownership" to mean the relevant set of rules doesn't work out, because the rules are different each time (we transfer the deleter in different ways, and may or may not reset the destination first). Let's go with this patch or something like it.

File: source/utilities.tex (hunk 1 above)
Comment (tkoeppe) [reply source/utilities.tex]: @jensmaurer: ping

Comment (jensmaurer) [reply source/utilities.tex]: Fixed.
